### PR TITLE
Add support for complex shadows

### PR DESCRIPTION
**Pull request by joelcmk**
Fixes issue #250

### CODE REVIEW

File: packages/core/src/components/Shadows/ShadowsSection.tsx [hunk 1]
```diff
@@ -177,6 +211,8 @@ export function ShadowsSection({
     onClose: onAddVariantModalClose,
   } = useDisclosure()
 
+  console.log(shadows)
```
Comment (alexdanilowicz) [inline packages/core/src/components/Shadows/ShadowsSection.tsx:214]:
```suggestion
```

**Comment by alexdanilowicz** [comment]
@joelcmk — super nice work on this! Really good.

1. Wanna fix up the merge conflicts by pulling main?
2. @ktra99 FYI on this PR
3. I think a space is missing after adding a shadow (should be `, ` aka comma space) 
4. There's an edge case if you add too many shadows: 

Screen Shot 2023-04-15 at 6 16 31 PM

5. looks like there's a bug where you can't actually edit the value input box. not sure if that's intentional but i feel like expected behavior is being able to copy paste in anything into the input or 
 edit any part of it: inhttps://www.loom.com/share/7d52a94c58bf4b3f8135279b0637d733

**Comment by joelcmk** [comment]
Hey Alex, thanks for the feedback. I'll work on those issues, just have a question about number 4. Should there be a limit for the number of shadows allowed?

**Comment by alexdanilowicz** [comment]
> Should there be a limit for the number of shadows allowed?

Ideally, no, as I don't think there's an actual limit with CSS, but honestly i'd be shocked if we get any users with more than 4.  so id do whatever is easiest and leads to the least code bloat: limit of 4? 

If you want to be fancy, you could have arrows that show you 4 at a time and then it's infinite (no limit) Also, I think maybe you should number them if possible? Like the icon for 1 has a 1 near it or on top of it, 2, 3, 4....

^ that can be a fast follow though. I'd rather have an actual user say "i need more than 4 shadows".

Main thing right now is just making sure it doesn't look/appear buggy.

**Comment by alexdanilowicz** [comment]
@joelcmk great work.

1. We might wanna do a regex to validate it's a valid box shadow? for example, you can input `a`. 

Screen Shot 2023-04-19 at 9 20 09 AM


2. so NICE work booking up Slider -> Input Box. ideally the input box to the slider values are also hooked up, otherwise it's kinda weird and feels buggy. Like if i change input box value to be "15px"for blur, it should change blur slider to 15px.

 
Screen Shot 2023-04-19 at 9 21 29 AM

3. I think the input box should only show one at a time. because the concept of "selecting a shadow" at the top suggests to me I'm only editing one shadow at a time. I think this will make 1 and 2 easier if you know the input box only has one shadow in it at a given time... (ill be honest, I'm not 100% sure because i would like user feedback to see if the user wants an input box to input all shadows at once.... but right now it feels like selecting one shadow at a time is confusing when the input box shows all of them.) 

Screen Shot 2023-04-19 at 9 23 35 AM

**Comment by alexdanilowicz** [comment]
Hi, this is looking better and better each time i see it:

1. I think ideally it doesn't tell you its an invalid format until you save. Its kinda of weird to type normally and it yells at you its invalid. Think about how password checkers work. They usually don't tell you its too short or too long until after you submit. 
Screen Shot 2023-05-01 at 4 19 52 PM

2. If it's invalid, it shouldn't also say "please fill out all the fields". Also, seems like both error reds are a different color! 
Screen Shot 2023-05-01 at 4 22 02 PM

3. on that note, "Please fill out all fields".. it should highlight the name form red if that's missing, otherwise its confusing whats missing. like, what's the error? it's unclear.
Screen Shot 2023-05-01 at 4 27 58 PM


3. let's make the save match the color (yours is blue, should be grey) and placement (save to the right, delete to the left) of the color modal
Screen Shot 2023-05-01 at 4 24 03 PM
Screen Shot 2023-05-01 at 4 23 35 PM

4. it's still kind of confusing to me what these are. I think they should be labeled: 1, 2, 3, 4... ask your partner or friend if it makes sense to them what it is. 
Screen Shot 2023-05-01 at 4 25 18 PM

5. Instead of variant name, it should say shadow name. Also, it should match the size of the header in the color modal. "Variant name" is much larger in the color modal than your modal.

great work! it's so close. and these are simple things.

**Pull request merged**